calculator/main.py

```diff
@@ -26,7 +31,22 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
+    if operation not in calc['operations'].keys():
+        raise InvalidOperation("{} operation not supported.".format(operation))
+    elif params == False:
+        raise InvalidParams("Given params are invalid.")
+    else: 
+        for i in params:
+            if isinstance(i, bool) or isinstance(i, str):
+                raise InvalidParams("Given params are invalid.")
+        
```

Comment:
Instead of 2 separate tests, you could have used an `or`.
```python
 elif not params or not all(isinstance(x, (int,float)) for x in params):
        raise InvalidParams('Given params are invalid.')
```
As you can see, I prefer raising the exception if params ARE NOT int or flat instead of raising if they ARE bool or str. What if one of the params is a tuple or a dict?